linkcheck/command/setup_config.py

```diff
@@ -204,8 +202,11 @@ def setup_config(config, options):
     if options.useragent is not None:
         config["useragent"] = options.useragent
     if options.cookiefile is not None:
-        if fileutil.is_readable(options.cookiefile):
-            config["cookiefile"] = options.cookiefile
-        else:
+        if not fileutil.is_valid_config_source(options.cookiefile):
+            log.error(
+                LOG_CMDLINE, _("Cookie file %r does not exist."), options.cookiefile)
+        elif not fileutil.is_readable(options.cookiefile):
```

Comment:
I did wonder while reading this code earlier, why log instead of terminating for invalid or unreadable config and cookie files? It feels like if the user explicitly specifies a cookie or config file and we were unable to use it, we should probably be as noisy as possible about that. Otherwise it can easily get lost in other things being printed.